### PR TITLE
Add all Voucher and Voucher+ holiday stops

### PR DESCRIPTION
**Pull request by mario-galic**
- Process all Voucher variants including Voucher+ variants using a single voucher model
- Adds both processor and API (api was done by @frj)
- Tested end-to-end in DEV [A-S00051849](https://gnmtouchpoint--dev.cs82.my.salesforce.com/a2k3E000000pLW4), [A-S00051850](https://gnmtouchpoint--dev.cs82.my.salesforce.com/a2k3E000000pLTU), A-S00051852, A-S00051853, A-S00051854, A-S00051855, [A-S00051858](https://gnmtouchpoint--dev.lightning.force.com/lightning/r/Holiday_Stop_Request__c/a2k3E000000pLZwQAM/view)
- Tested end-to-end in UAT [A-S00080144](https://gnmtouchpoint--uat.cs82.my.salesforce.com/a2k3E000000pLWYQA2)

### CODE REVIEW

**Comment by mario-galic** [comment]
[Add Weekend, Sixday and Everyday holiday stops](https://trello.com/c/pAlW331X/502-add-weekend-sixday-and-everyday-holiday-stops)

File: lib/holiday-stops/src/main/scala/com/gu/salesforce/holiday_stops/SalesforceHolidayStopRequestsDetail.scala [hunk 1]
```diff
@@ -151,19 +126,19 @@ object SalesforceHolidayStopRequestsDetail extends Logging {
...
         .runRequestMultiArg
 
-    def toRequest(productKey: ProductRatePlanKey, date: LocalDate): GetRequestWithParams = {
+    def toRequest(product: Product, date: LocalDate): GetRequestWithParams = {
       val soqlQuery = s"""
                          | $SOQL_SELECT_CLAUSE
                          | FROM $holidayStopRequestsDetailSfObjectRef
-                         | WHERE Product_Name__c LIKE '${productKey.productType.value}%'
-                         | AND Holiday_Stop_Request__r.SF_Subscription__r.Rate_Plan_Name__c = '${productKey.ratePlanName.value}'
+                         | WHERE Product_Name__c LIKE 'Newspaper Voucher%'
```
Comment (kelvin-chappell) [inline lib/holiday-stops/src/main/scala/com/gu/salesforce/holiday_stops/SalesforceHolidayStopRequestsDetail.scala]: Perhaps use `= 'Newspaper Voucher'` instead as the product name is identical for all cases.

File: lib/holiday-stops/src/test/scala/com/gu/holiday_stops/HolidayCreditTest.scala [hunk 2]
```diff
@@ -2,35 +2,38 @@ package com.gu.holiday_stops
...
 import java.time.LocalDate
 
-import com.gu.holiday_stops.subscription.{CurrentGuardianWeeklySubscription, GuardianWeeklyHolidayCredit, RatePlan}
+import com.gu.holiday_stops.subscription.{GuardianWeeklySubscription, RatePlan}
+import com.gu.salesforce.holiday_stops.SalesforceHolidayStopRequestsDetail.StoppedPublicationDate
 import org.scalatest.{EitherValues, FlatSpec, Matchers}
 
 class HolidayCreditTest extends FlatSpec with Matchers with EitherValues {
 
+  val stoppedPublicationDate = StoppedPublicationDate(LocalDate.parse("2019-09-02").minusDays(1))
```
[the remark this answers is not in the document]
Comment (mario-galic) [reply lib/holiday-stops/src/test/scala/com/gu/holiday_stops/HolidayCreditTest.scala]: I kind of want to emphasise it is one day before so I am being extra explicit.

File: lib/holiday-stops/src/main/scala/com/gu/holiday_stops/ActionCalculator.scala [hunk 3]
```diff
@@ -108,19 +108,64 @@ object ActionCalculator {
...
+  val EverydayPlusVoucherSuspensionConstants = EverydayVoucherSuspensionConstants
+  val SixdayPlusVoucherSuspensionConstants = SixdayVoucherSuspensionConstants
+  val WeekendPlusVoucherSuspensionConstants = WeekendVoucherSuspensionConstants
+  val SundayPlusVoucherSuspensionConstants = SundayVoucherSuspensionConstants
+  val SaturdayPlusVoucherSuspensionConstants = SaturdayVoucherSuspensionConstants
+
+  def voucherSuspensionConstans(issueSuspensionConstants: List[IssueSuspensionConstants]) =
+    SuspensionConstants(issueSuspensionConstants.size * 6, issueSuspensionConstants)
+
+  lazy val VoucherProcessorLeadTime: Int = 1
```
[the remark this answers is not in the document]
Comment (mario-galic) [reply lib/holiday-stops/src/main/scala/com/gu/holiday_stops/ActionCalculator.scala]: Strangely it makes the test fail if I remove lazy.

**Comment by mario-galic** [comment]
[GOAL : Release Sunday product variant for holiday stop to CSRs](https://trello.com/c/kxFA7h3k/472-goal-release-sunday-product-variant-for-holiday-stop-to-csrs)

**Pull request merged**